src/Plugin/Condition/DataListContains.php

```diff
@@ -39,6 +39,20 @@ class DataListContains extends RulesConditionBase {
   /**
    * {@inheritdoc}
    */
+  public function refineContextDefinitions() {
+
+    // Refine the item context.
+    if ($item = $this->getContextValue('item')) {
+      if ($item instanceof EntityInterface && $type = $item->getEntityTypeId()) {
```

Comment:
why EntityInterface here? I think this should just be "$typed_data_item = $this->getContextData()" and then "$this->pluginDefinition['context']['item']->setDataType($typed_data_item->getDataDefinition()->getDataType());"